packages/lib-algorand/index.ts

```diff
@@ -0,0 +1,89 @@
+import { DfnsApiClient, DfnsError } from '@dfns/sdk'
+import { GetWalletResponse, GenerateSignatureResponse } from '@dfns/sdk/types/wallets'
+
+import { encodeObj, Transaction, EncodedSignedTransaction, decodeObj } from 'algosdk'
+export type DfnsWalletOptions = {
+  walletId: string
+  dfnsClient: DfnsApiClient
+}
+
+const hexToBuffer = (hex: string): Buffer => {
+  return Buffer.from(hex.replace(/^0x/, ''), 'hex')
+}
+
+export const singleSigTransactionToEncoded = (tx: Transaction): EncodedSignedTransaction =>{
+  return {
+    txn: tx.get_obj_for_encoding()
+  }
+}
+
+type WalletMetadata = GetWalletResponse
+
+const assertSigned = (res: GenerateSignatureResponse) => {
+  if (res.status === 'Failed') {
+    throw new DfnsError(-1, 'signing failed', res)
+  } else if (res.status !== 'Signed') {
+    throw new DfnsError(
+      -1,
+      'cannot complete signing synchronously because this wallet action requires policy approval',
+      res
+    )
+  }
+}
+
+export class DfnsWallet {
+  private readonly dfnsClient: DfnsApiClient
+
+  private constructor(private metadata: WalletMetadata, options: DfnsWalletOptions) {
+    this.dfnsClient = options.dfnsClient
+  }
+
+  public static async init(options: DfnsWalletOptions) {
+    const { walletId, dfnsClient } = options
+    const res = await dfnsClient.wallets.getWallet({ walletId })
+
+    if (res.status !== 'Active') {
+      throw new DfnsError(-1, 'wallet not active', { walletId, status: res.status })
+    }
+
+    const { scheme, curve } = res.signingKey
+    if (scheme !== 'EdDSA') {
+      throw new DfnsError(-1, 'key scheme is not EdDSA', { walletId, scheme })
+    }
+    if (curve !== 'ed25519') {
+      throw new DfnsError(-1, 'key curve is not ed25519', { walletId, curve })
+    }
+
+    if (res.network !== 'Algorand' && res.network !== 'AlgorandTestnet') {
+      throw new DfnsError(-1, 'wallet is not bound to Algorand or AlgorandTestnet', {
+        walletId,
+        network: res.network,
+      })
+    }
+
+    return new DfnsWallet(res, options)
+  }
+
+  public get address(): string {
+    return this.metadata.address!
+
+  }
+
+  // sign single signature transaction
+  public async signTransaction(transaction: EncodedSignedTransaction): Promise<EncodedSignedTransaction> {
```

Comment:
would be nice to add an overload for `public async signTransaction(transaction: Transaction)` where we will handle the wrapping.

Reply:
That's what we talked about, but instead, I provide a helper function `singleSigTransactionToEncoded`, so they can call signTransaction afterwards. Do you still prefer having the overload?

Reply:
yeah just a bit nicer have it just work without extra explanations.